perf(ivy): store views directly on LContainer

crisbeto opened the pull request:
Stores the views that are part of a container directly on the `LContainer`, rather than maintaining a dedicated sub-array.

This PR resolves FW-1288.

kara commented on packages/core/src/render3/view_engine_compatibility.ts:
```diff
@@ -201,14 +201,19 @@ export function createContainerRef(
 
       clear(): void {
-        while (this._lContainer[VIEWS].length) {
+        while (this._lContainer.length - CONTAINER_HEADER_OFFSET > 0) {
```
NIt: readability

```suggestion
        while (this.length) {
```

kara reviewed (approved):
LGTM, one nit

kara commented:
[presubmit](http://test/OCL:245793679:BASE:245793705:1556564346947:e5becfc5)

crisbeto committed:
perf(ivy): store views directly on LContainer

Stores the views that are part of a container directly on the `LContainer`, rather than maintaining a dedicated sub-array.

This PR resolves FW-1288.
